Mathlib/RingTheory/Polynomial/Selmer.lean

```diff
@@ -75,4 +279,173 @@ theorem X_pow_sub_X_sub_one_irreducible_rat (hn1 : n ≠ 1) : Irreducible (X ^ n
+    (E₁ E₂ : Type*) [Field E₁] [Algebra F E₁] [Field E₂] [Algebra F E₂]
+    [Fact (p.Splits (algebraMap F E₁))] [Fact (p.Splits (algebraMap F E₂))] :
+    Gal.galActionHom p E₁ =
+      ((Polynomial.Gal.rootsEquivRoots p E₂).symm.trans
+        (Polynomial.Gal.rootsEquivRoots p E₁)).permCongrHom.toMonoidHom.comp
+        (Gal.galActionHom p E₂)
+       := by
+  ext
+  simp [permCongrHom, Perm.permCongrHom, Gal.galActionHom, Polynomial.Gal.smul_def]
+
+attribute [-instance] Polynomial.Gal.galActionAux -- should be local to PolynomialGaloisGroup.lean
+
+theorem X_pow_sub_X_sub_one_gal :
+    Function.Bijective (Gal.galActionHom (X ^ n - X - 1 : ℚ[X]) ℂ) := by
+  classical
+  let f : ℚ[X] := X ^ n - X - 1
+  let K := f.SplittingField
+  have : Fact (f.Splits (algebraMap ℚ K)) := ⟨SplittingField.splits f⟩
+  have : NumberField K := by constructor
+  have : IsGalois ℚ K := by constructor
+  let R := 𝓞 K
+  let G := f.Gal
+  suffices Function.Bijective (Gal.galActionHom f K) by
+    rw [switchinglemma f ℂ K]
+    exact (((Gal.rootsEquivRoots f f.SplittingField).symm.trans
+      (Gal.rootsEquivRoots f ℂ)).permCongrHom.toEquiv.comp_bijective _).mpr this
+  have : MulAction.IsPretransitive G (f.rootSet K) := by
+    rcases eq_or_ne n 1 with rfl | hn
+    · have : IsEmpty (rootSet f K) := by simp [f]
+      infer_instance
+    exact Gal.galAction_isPretransitive _ _ (X_pow_sub_X_sub_one_irreducible_rat hn)
+  let _ : MulSemiringAction G R := IsIntegralClosure.MulSemiringAction ℤ ℚ K R
+  let S : Set G := ⋃ (q : Ideal R) (hq : q.IsMaximal),
+    ((↑(AddSubgroup.inertia q.toAddSubgroup G : Subgroup G) : Set G) \ {1})
+  have hS1 : Subgroup.closure S = ⊤ := by
+    simp only [S, Subgroup.closure_iUnion, Subgroup.closure_eq, Subgroup.closure_diff_one]
+    exact keythm K
+  suffices hS2 : ∀ σ ∈ S, Perm.IsSwap (MulAction.toPermHom f.Gal (f.rootSet K) σ) by
+    exact ⟨Gal.galActionHom_injective f K, surjective_of_isSwap_of_isPretransitive S hS2 hS1⟩
+  simp only [S, Set.mem_diff, Set.mem_iUnion]
+  rintro σ ⟨q, hq, hσ, hσ1 : σ ≠ 1⟩
+  let H : Subgroup G := inertiaSubgroup q
+  let F := R ⧸ q
+  let π : R →+* F := Ideal.Quotient.mk q
+  have : Field F := Ideal.Quotient.field q
+  have : MulAction H (f.rootSet K) := inferInstance
+  have : MulAction H F := sorry
+  let ϕ : f.rootSet K → F := sorry -- equivariant map under which σ acts trivially
+  let f' : F[X] := X ^ n - X - 1
+
+
+  clear hS1 S
+
+  -- σ ∈ inertiaSubgroup q
+  -- σ acts trivially on the
+
+  -- finite field, might not need to consider the characteristic
+  -- reduce to action on roots in R
+  sorry
+
+  -- have : ∀ p : Nat.Primes, ∀ q : factors (map (algebraMap ℤ R) p)
+  -- roots lie in the ring of integers OK
+  -- if q is a prime idea of OK, then there is a ring homomorphism to the finite field OK/q
+  -- the whole Galois group acts on OK
+  -- the decomposition group acts on OK/q
+  -- the inertia group acts trivially on OK/q
+  --
+  -- there are n roots in OK
```

Comment:
The same argument also works in the geometric context, for galois groups over Q(X) (see Serre Topics in Galois theory, §4.4) and I hope it will be possible to set up such a framework.